### PR TITLE
Add blog about Camel 3.4 whats new

### PR DESCRIPTION
**Pull request by davsclaus**
Wait until 3.4 is GA.

Review and feedback is welcome

### CODE REVIEW

**Comment by omarsmak** [comment]
Would it also make sense to mention about the new maven plugin for the 3rd party component developers? https://camel.apache.org/manual/latest/camel-component-maven-plugin.html

**Comment by davsclaus** [comment]
> Would it also make sense to mention about the new maven plugin for the 3rd party component developers? https://camel.apache.org/manual/latest/camel-component-maven-plugin.html

Yeah sure. However I do think its most worthwhile to add a section about it in the 3.4 upgrade guide.

File: content/blog/2020/06/camel34-whatsnew/index.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+For more details about LTS vs non-LTS releases see this [blog post](https://camel.apache.org/blog/LTS-Release-Schedule/).
+
+
+### So whats in this release?
+
+This release is mostly a more robust and bug fix release. 
+
+We have also continued the work to make Camel more modular and lighter. 
+This time we removed the needed for JAXB in the swagger and openapi modules.
```
Comment (gzurowski) [inline content/blog/2020/06/camel34-whatsnew/index.md:21]: "removed the need for JAXB..." or maybe just "removed the dependency on JAXB..."

File: content/blog/2020/06/camel34-whatsnew/index.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+This helps Camel on GraalVM and native compilation as JAXB is a heavy piece of stack,
+allowing GraalVM to eliminate it more easily.
+
+We continued to remove usage of reflection in Camel and found a few spots more where
+reflection was in use, when configuring nested options.
+
+We also added back support for configuring duration values using the shorthand syntax, 
+such as `timeout=30000` can be specified as `timeout=30s`. We had to remove this in earlier
+versions of Camel 3 due to optimizations. But for Camel 3.4 we wound a new way.
```
Comment (gzurowski) [inline content/blog/2020/06/camel34-whatsnew/index.md:30]: "_found_ a new way"

File: content/blog/2020/06/camel34-whatsnew/index.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+We continued to remove usage of reflection in Camel and found a few spots more where
+reflection was in use, when configuring nested options.
+
+We also added back support for configuring duration values using the shorthand syntax, 
+such as `timeout=30000` can be specified as `timeout=30s`. We had to remove this in earlier
+versions of Camel 3 due to optimizations. But for Camel 3.4 we wound a new way. 
+
+#### Supervising route controller
+
+The work on the supervising route controller is complete. When Camel startup the default route controller
```
Comment (gzurowski) [inline content/blog/2020/06/camel34-whatsnew/index.md:34]: "During Camel startup..." or "When Camel starts up..."

File: content/blog/2020/06/camel34-whatsnew/index.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+is handling starting the routes safely. The default strategy is that if a route fails to startup then Camel
+itself will also fail its startup (fail fast). 
+
+The supervising route controller is a different strategy that allows to startup routes independent from Camel itself.
+This new controller will startup the routes using a background task that can re-scheduled routes that
```
Comment (gzurowski) [inline content/blog/2020/06/camel34-whatsnew/index.md:39]: "...that can _reschedule_ routes..."

File: content/blog/2020/06/camel34-whatsnew/index.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+#### Health Check
+
+We have reworked Camel's health-check, to work similar across runtimes, whether its standalone, Spring Boot,
+Camel-K, or Quarkus.
+
+We also introduced the concept of readiness and liveness so a health check can be either one or both.
+Each health check can be configured, from `application.properties` the same way, and its all reflection free.
+
+The previous mentioned examples also comes with health-check so make sure to check those.
```
Comment (gzurowski) [inline content/blog/2020/06/camel34-whatsnew/index.md:56]: "The _previously_ mentioned..."

File: content/blog/2020/06/camel34-whatsnew/index.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+```
+
+#### Other Changes
+
+You can now configure Camel's thread pool (profiles) the same way for standalone, Camel K, Camel Quarkus and Spring Boot.
+
+Some of the components (more to come in the future) we have moved initialization logic to an earlier phase when possible
+which allows these components to initialize at build time, which makes Camel startup faster (especially for GraalVM or Quarkus runtimes).
+
+This release supports Spring Boot 2.3.
```
Comment (gzurowski) [inline content/blog/2020/06/camel34-whatsnew/index.md:87]: I assume many people will seek for this detail, maybe we should put it into it's own (short) section for making it more prominent?

File: content/blog/2020/06/camel34-whatsnew/index.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
+---
+title: "Apache Camel 3.4 What's New"
+date: 2020-06-19
+authors: [davsclaus]
+categories: ["Releases"]
+preview: Whats included in the Camel 3.4 LTS release.
```
Comment (zregvart) [inline content/blog/2020/06/camel34-whatsnew/index.md:6]:
```suggestion
preview: The long term release version 3.4 of Camel is here, find out what's new.
draft: false
```

File: content/blog/2020/06/camel34-whatsnew/index.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+Apache Camel 3.4 is the first LTS (Long Term Support) release of Camel 3.
+
+This release will be active supported with regular patch releases (important bug and security fixes) for 1-year.
+
+For more details about LTS vs non-LTS releases see this [blog post](https://camel.apache.org/blog/LTS-Release-Schedule/).
+
+
+### So whats in this release?
+
+This release is mostly a more robust and bug fix release.
```
Comment (zregvart) [inline content/blog/2020/06/camel34-whatsnew/index.md:18]:
```suggestion
This release is mostly about robustness and bug fixes. 
```

File: content/blog/2020/06/camel34-whatsnew/index.md [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+title: "Apache Camel 3.4 What's New"
+date: 2020-06-19
+authors: [davsclaus]
+categories: ["Releases"]
+preview: Whats included in the Camel 3.4 LTS release.
+---
+
+Apache Camel 3.4 is the first LTS (Long Term Support) release of Camel 3.
+
+This release will be active supported with regular patch releases (important bug and security fixes) for 1-year.
```
Comment (zregvart) [inline content/blog/2020/06/camel34-whatsnew/index.md:11]:
```suggestion
This release will be actively supported with regular patch releases containing important bug and security fixes for 1-year.
```

File: content/blog/2020/06/camel34-whatsnew/index.md [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+
+For more details about LTS vs non-LTS releases see this [blog post](https://camel.apache.org/blog/LTS-Release-Schedule/).
+
+
+### So whats in this release?
```
Comment (zregvart) [inline content/blog/2020/06/camel34-whatsnew/index.md:16]:
```suggestion
### So what's in this release?
```

File: content/blog/2020/06/camel34-whatsnew/index.md (hunk 1 above)
Comment (zregvart) [inline content/blog/2020/06/camel34-whatsnew/index.md:21]:
```suggestion
This time we removed the dependency on JAXB in the Swagger and OpenAPI modules.
```

File: content/blog/2020/06/camel34-whatsnew/index.md [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+
+This release is mostly a more robust and bug fix release. 
+
+We have also continued the work to make Camel more modular and lighter. 
+This time we removed the needed for JAXB in the swagger and openapi modules.
+This helps Camel on GraalVM and native compilation as JAXB is a heavy piece of stack,
+allowing GraalVM to eliminate it more easily.
```
Comment (zregvart) [inline content/blog/2020/06/camel34-whatsnew/index.md:23]:
```suggestion
With this, native compilation of Camel on GraalVM is faster and more lightweight.
```

File: content/blog/2020/06/camel34-whatsnew/index.md [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+
+We continued to remove usage of reflection in Camel and found a few spots more where
+reflection was in use, when configuring nested options.
```
Comment (zregvart) [inline content/blog/2020/06/camel34-whatsnew/index.md:26]:
```suggestion
We continued to remove usage of reflection in Camel and found a couple of spots more where
reflection was used. For example when configuring nested options.
```

File: content/blog/2020/06/camel34-whatsnew/index.md [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+such as `timeout=30000` can be specified as `timeout=30s`. We had to remove this in earlier
+versions of Camel 3 due to optimizations. But for Camel 3.4 we wound a new way. 
+
+#### Supervising route controller
+
+The work on the supervising route controller is complete. When Camel startup the default route controller
+is handling starting the routes safely. The default strategy is that if a route fails to startup then Camel
+itself will also fail its startup (fail fast). 
+
+The supervising route controller is a different strategy that allows to startup routes independent from Camel itself.
```
Comment (zregvart) [inline content/blog/2020/06/camel34-whatsnew/index.md:38]:
```suggestion
The supervising route controller is a different strategy that allows starting routes independent from Camel itself.
```

File: content/blog/2020/06/camel34-whatsnew/index.md [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+
+We also added back support for configuring duration values using the shorthand syntax, 
+such as `timeout=30000` can be specified as `timeout=30s`. We had to remove this in earlier
+versions of Camel 3 due to optimizations. But for Camel 3.4 we wound a new way. 
+
+#### Supervising route controller
+
+The work on the supervising route controller is complete. When Camel startup the default route controller
+is handling starting the routes safely. The default strategy is that if a route fails to startup then Camel
+itself will also fail its startup (fail fast).
```
Comment (zregvart) [inline content/blog/2020/06/camel34-whatsnew/index.md:36]:
```suggestion
The work on the supervising route controller is complete. When Camel starts up the default route controller
is performing safe startup of routes. The default strategy is that if a route fails to startup then Camel
itself will also fail to startup (fail fast). 
```

File: content/blog/2020/06/camel34-whatsnew/index.md [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+
+You can find more details in the [Route Controller](https://camel.apache.org/manual/latest/route-controller.html) documentation.
+
+#### Health Check
+
+We have reworked Camel's health-check, to work similar across runtimes, whether its standalone, Spring Boot,
+Camel-K, or Quarkus.
+
+We also introduced the concept of readiness and liveness so a health check can be either one or both.
+Each health check can be configured, from `application.properties` the same way, and its all reflection free.
```
Comment (zregvart) [inline content/blog/2020/06/camel34-whatsnew/index.md:54]:
```suggestion
Each health check can be configured, from `application.properties` the same way, and its all reflection-free.
```

File: content/blog/2020/06/camel34-whatsnew/index.md (hunk 5 above)
Comment (zregvart) [inline content/blog/2020/06/camel34-whatsnew/index.md:56]:
```suggestion
The examples above also show how to configure health and liveness checks, so be sure to check out those.
```

File: content/blog/2020/06/camel34-whatsnew/index.md [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+
+The previous mentioned examples also comes with health-check so make sure to check those.
+See more details in the [Health Check](https://camel.apache.org/manual/latest/health-check.html) documentation.
+
+### Endpoint DSL
+
+The [Endpoint DSL](https://camel.apache.org/manual/latest/Endpoint-dsl.html) had a number of annoying bugs fixed and other improvements.
```
Comment (zregvart) [inline content/blog/2020/06/camel34-whatsnew/index.md:61]:
```suggestion
The [Endpoint DSL](https://camel.apache.org/manual/latest/Endpoint-dsl.html) had several annoying bugs fixed along with other improvements.
```

File: content/blog/2020/06/camel34-whatsnew/index.md [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
+Its now also easier to use Endpoint DSL to configure endpoints in POJOs as Java fields in a type safe manner,
```
Comment (zregvart) [inline content/blog/2020/06/camel34-whatsnew/index.md:62]:
```suggestion
It is now also easier to use Endpoint DSL to configure endpoints in POJOs as Java fields in a type-safe manner, 
```

File: content/blog/2020/06/camel34-whatsnew/index.md [hunk 18]
```diff
@@ -0,0 +1,95 @@
...
+Each health check can be configured, from `application.properties` the same way, and its all reflection free.
+
+The previous mentioned examples also comes with health-check so make sure to check those.
+See more details in the [Health Check](https://camel.apache.org/manual/latest/health-check.html) documentation.
+
+### Endpoint DSL
+
+The [Endpoint DSL](https://camel.apache.org/manual/latest/Endpoint-dsl.html) had a number of annoying bugs fixed and other improvements.
+Its now also easier to use Endpoint DSL to configure endpoints in POJOs as Java fields in a type safe manner, 
+and which can be used with `FluentProducerTemplate` or in your `RouteBuilder` classes as shown:
```
Comment (zregvart) [inline content/blog/2020/06/camel34-whatsnew/index.md:63]:
```suggestion
by using `FluentProducerTemplate` and in `RouteBuilder` classes as in this example:
```

File: content/blog/2020/06/camel34-whatsnew/index.md [hunk 19]
```diff
@@ -0,0 +1,95 @@
...
+
+    private final EndpointProducerBuilder mqtt = paho("sensor").clientId("myClient").userName("scott").password("tiger");
+
+    public void sendToSensor(String data) {
+        producer.withBody(data).to(mqtt).send();
+    }
+
+}    
+```
+
```
Comment (zregvart) [inline content/blog/2020/06/camel34-whatsnew/index.md:79]:
```suggestion

You can read more about Endpoint DSL [in the manual](/manual/latest/Endpoint-dsl.html).
```

File: content/blog/2020/06/camel34-whatsnew/index.md [hunk 20]
```diff
@@ -0,0 +1,95 @@
...
+#### Other Changes
+
+You can now configure Camel's thread pool (profiles) the same way for standalone, Camel K, Camel Quarkus and Spring Boot.
```
Comment (zregvart) [inline content/blog/2020/06/camel34-whatsnew/index.md:82]:
```suggestion
You can now configure Camel's thread pool (profiles) the same way for standalone, Camel K, Camel Quarkus, and Spring Boot.
```

File: content/blog/2020/06/camel34-whatsnew/index.md [hunk 21]
```diff
@@ -0,0 +1,95 @@
...
+
+Some of the components (more to come in the future) we have moved initialization logic to an earlier phase when possible
+which allows these components to initialize at build time, which makes Camel startup faster (especially for GraalVM or Quarkus runtimes).
+
+This release supports Spring Boot 2.3.
+
+Only two new components was added to this release:
```
Comment (zregvart) [inline content/blog/2020/06/camel34-whatsnew/index.md:89]:
```suggestion
This release also adds two new components:
```

**Comment by davsclaus** [comment]
Thanks for the feedback. The suggestion has been applied. And if you find any typo or grammar improvement then feel free to edit/commit directly.

**Comment by omarsmak** [comment]
> > Would it also make sense to mention about the new maven plugin for the 3rd party component developers? https://camel.apache.org/manual/latest/camel-component-maven-plugin.html
> 
> Yeah sure. However I do think its most worthwhile to add a section about it in the 3.4 upgrade guide.

I have added it in the migration guide

**Comment by davsclaus** [comment]
> > Would it also make sense to mention about the new maven plugin for the 3rd party component developers? https://camel.apache.org/manual/latest/camel-component-maven-plugin.html
> 
> Yeah sure. However I do think its most worthwhile to add a section about it in the 3.4 upgrade guide.

What you added was great so I copied it over in this blog post

**Pull request merged**